### PR TITLE
Drop unneeded escaping of - (fixes #3379)

### LINKED ISSUE #3379: New version of grep produces warnings
grep-3.8-i586-1.txz from Slackware Current

During build:
```
# chroot rootfs-complete/
grep: warning: stray \ before -
grep: warning: stray \ before -
grep: warning: stray \ before -

```
In running system when either LxTerminal or URXVT start:
Screenshot1

Screenshot2

### PR DESCRIPTION
**Pull request by dimkr**
Alternative to #3395 that _should_ work.

### CODE REVIEW

File: woof-code/rootfs-skeleton/usr/local/petget/dependencies.sh [hunk 1]
```diff
@@ -66,14 +66,14 @@ DB_OTHERS="`ls -1 ${PREPATH}Packages-* | grep -v "$DB_MAIN"`"
 #120903 improve pkg db selection...
 case $DB_MAIN in
   *"-puppy-2-"*|*"-puppy-3-"*|*"-puppy-4-"*|*"-puppy-5-"*) DB_OTHERS="" ;;
-  *) DB_OTHERS="`echo "$DB_OTHERS" | grep -v '\\-puppy\\-[2345]\\-'`" ;; #do not look in puppy-2, puppy-3, puppy-4 or puppy-5.
+  *) DB_OTHERS="`echo "$DB_OTHERS" | grep -v '\-puppy-[2345]-'`" ;; #do not look in puppy-2, puppy-3, puppy-4 or puppy-5.
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/local/petget/dependencies.sh:69]: Ha, this one never worked in first place!

```
~$ (echo abcd; echo a-puppy-4-) | grep -v '\\-puppy\\-[2345]\\-'
abcd
a-puppy-4-
~$ (echo abcd; echo a-puppy-4-) | grep -v '\-puppy-[2345]-'
abcd
```

**Review by 01micko — COMMENTED** [review]
SO, by the rationale in these changes, only a leading `-` needs escaping in a `grep` call.. yes?

**Comment by dimkr** [comment]
> SO, by the rationale in these changes, only a leading `-` needs escaping in a `grep` call.. yes?

Yes - either escape the first `-` (i.e. `grep '\--show-progress'`), or put `--` before the pattern (i.e. `grep -- '--show-progress'`).

**Review by peabee — APPROVED** [review]
OK - will withdraw #3395 and close #3379

**Comment by dimkr** [comment]
> OK - will withdraw #3395 and close #3379

Let's wait with closing #3379 - first of all, see if this PR works for you at all. Once it's merged, the issue can be closed.

**Review by 01micko — APPROVED** [review]

**Comment by dimkr** [comment]
@peabee I'm waiting for you to confirm that this solves your problem. You're probably the only one running woof-CE on Slackware current.

**Comment by peabee** [comment]
Its not easy to test until it has been merged into testing branch............??

I guess I have to download Woof-CE from https://github.com/dimkr/woof-CE/tree/bugfix/more-grep-warnings

**Comment by peabee** [comment]
Just done manual edits of 0setup and 1download and they seemed OK - no grep warnings.

**Comment by dimkr** [comment]
> Its not easy to test until it has been merged into testing branch............??

It wasn't easy to fix either 😠

**Comment by 01micko** [comment]
I'll be building on -current soon enough too. My wayland -current install even prints! :astonished: 

I reckon merge this. In fact Ill do it now, deal with consequences, if any, later.

**Comment by peabee** [comment]
Test build:
BUILD_FROM_WOOF='testing;e75c078e3;2022-09-13 19:20:00 +1000'

Still get a warning during 3builddistro:

> Updating system config..
> gtk-update-icon-cache: Cache file created successfully.
> gtk-update-icon-cache: Cache file created successfully.
> gtk-update-icon-cache: Cache file created successfully.
> gtk-update-icon-cache: The generated cache was invalid.
> gtk-update-icon-cache: Cache file created successfully.
> gtk-update-icon-cache: Cache file created successfully.
> Generating /root/.jwmrc...
> Executing depmod, to update module files...
> grep: warning: stray \ before -
> 
> Setting Desktop Icon Theme...
>

**Comment by peabee** [comment]
OK after a lot of digging looks like its:
/woof-code/rootfs-skeleton/usr/sbin/quicksetup
line 458:
`  [ "`echo -n "$CURRLANG" | grep 'UTF\\-8'`" != "" ] && UTF8='.UTF-8'
`

**Pull request merged**